lib/checkcondition.cpp

```diff
@@ -903,6 +903,108 @@ static int countPar(const Token *tok1, const Token *tok2)
     return par;
 }
 
+// Clarify '[if|while|function] (x = foo() && bar())'
+void CheckCondition::clarifyAssignmentAndLogicalCondition()
+{
+    if (!_settings->isEnabled("style"))
+        return;
+
+    const SymbolDatabase *symbolDatabase = _tokenizer->getSymbolDatabase();
+    const std::size_t functions = symbolDatabase->functionScopes.size();
+    for (std::size_t i = 0; i < functions; ++i) {
+        const Scope * scope = symbolDatabase->functionScopes[i];
+        for (const Token* tok = scope->classStart->next(); tok != scope->classEnd; tok = tok->next()) {
+
+            const Token* root = tok;
+            if(!tok || tok->str() != "(") {
```

Comment:
yes it's a good idea to check if astOperand is NULL. The other NULL pointer checks are fine it was just this one that is redundant.